etl/steps/__init__.py

```diff
@@ -492,6 +508,47 @@ def _walden_dataset(self) -> WaldenDataset:
 
 
 @dataclass
+class SnapshotStep(Step):
+    path: str
+
+    def __init__(self, path: str) -> None:
+        self.path = path
+
+    def __str__(self) -> str:
+        return f"snapshot://{self.path}"
+
+    def run(self) -> None:
+        self.dvc_repo.pull(self._path)
+
+    def is_dirty(self) -> bool:
+        # run dvc status and cache it
+        with dvc_lock:
+            if "dvc_status" not in cache:
+                cache["dvc_status"] = self.dvc_repo.status()
+
+        # if path is in dvc status output, it's dirty
+        return self._dvc_path in cache["dvc_status"]
+
+    def has_existing_data(self) -> bool:
+        return True
+
+    def checksum_output(self) -> str:
+        with open(self._dvc_path, "r") as f:
+            return yaml.safe_load(f)["outs"][0]["md5"]
```

Comment:
The checksum of this step should be of the combination of the data and the metadata, which is what Walden steps do in `checksum_output()`.

Probably we should stuff metadata into the `.dvc` file itself and just return the checksum of that file here, instead of reporting the checksum of the binary file that DVC stored.

Reply:
> Probably we should stuff metadata into the .dvc file itself and just return the checksum of that file here, instead of reporting the checksum of the binary file that DVC stored.

Yeah, good idea. I was a bit worried about having metadata in `.dvc` file, but it feels good after some refactoring.